packages/react-broadcast/modules/components/configure/configure.tsx

```diff
@@ -22,36 +25,56 @@ type State = {
   configurationId?: string;
 };
 
+const initialAdapterStatus: TAdapterStatus = {
+  isReady: false,
+  subscriptionStatus: TAdapterSubscriptionStatus.Subscribed,
+};
+const initialFlags: TFlags = {};
+
 const Configure = <AdapterInstance extends TAdapter>(
   props: Props<AdapterInstance>
 ) => {
-  const [flags, setFlags] = React.useState<State['flags']>({});
-  const [status, setStatus] = React.useState<State['status']>({});
+  const [flags, setFlags] = React.useState<State['flags']>(initialFlags);
+  const [status, setStatus] = React.useState<State['status']>(
+    initialAdapterStatus
+  );
 
   // NOTE:
   //   Using this prevents the callbacks being invoked
   //   which would trigger a setState as a result on an unmounted
   //   component.
-  useAdapterSubscription(props.adapter);
+  const getHasAdapterSubscriptionStatus = useAdapterSubscription(props.adapter);
 
   const handleUpdateFlags = React.useCallback(
```

Comment:
Nit: you can specify the callback signature to `useCallback` instead of typing the function itself

```ts
React.useCallback<(foo: Foo) => void>(
  foo => {
    // foo is typed as Foo
  },
  []
)
```

Reply:
This I am actually never super sure about. I've read articles where people say type the function as TS is smart enough to "fill in the generic" and the other way round. I haven't decides which way I like more yet. You have opinions on it? I kind of also see value in having generics about types allowing you to have it not in arguments directly.

Reply:
Yeah sometimes the function is "simple enough" that TS can infer the signature. I usually check if that's the case, otherwise I'll add the type annotations.
I kind of prefer to pass the signature to `useCallback` as it makes it easier to see the signature at first glance, especially if the function is big.